src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java

```diff
@@ -0,0 +1,1463 @@
+
+        final AmountType3Choice paymentAmount = constructPaymentAmount(paymentDetails);
+        creditTransferTransactionInformation.setAmt(paymentAmount);
+
+        final PartyIdentification32 creditor = constructCreditor(templatePaymentGroup, extractTypeContext);
+        creditTransferTransactionInformation.setCdtr(creditor);
+
+        /*
+         * CU Customization: Added the ability to suppress remittances from payments that don't need it.
+         */
+        if (shouldAddRemittanceInformationToPayment(templatePaymentGroup, extractTypeContext)) {
+            final RemittanceInformation5 remittanceInformation =
+                    constructRemittanceInformation(templatePaymentGroup, paymentDetails, extractTypeContext);
+            creditTransferTransactionInformation.setRmtInf(remittanceInformation);
+        }
+
+        if (extractTypeContext.isExtractionType(ExtractionType.ACH)) {
+            final BranchAndFinancialInstitutionIdentification4 creditorAgent =
+                    constructCreditorAgent(templatePaymentGroup);
+            creditTransferTransactionInformation.setCdtrAgt(creditorAgent);
+
+            final CashAccount16 creditorAccount = constructCreditorAccount(templatePaymentGroup);
+            creditTransferTransactionInformation.setCdtrAcct(creditorAccount);
+
+            final PaymentTypeInformation19 paymentTypeInformation =
+                    constructPaymentTypeInformationWithLocalInstrument(templatePaymentGroup);
+            creditTransferTransactionInformation.setPmtTpInf(paymentTypeInformation);
+        } else if (extractTypeContext.isExtractionType(ExtractionType.CHECK)) {
+            final Cheque6 check = constructCheck(disbursementNumber, templatePaymentGroup);
+            creditTransferTransactionInformation.setChqInstr(check);
+        }
+
+        return creditTransferTransactionInformation;
+    }
+
+    /*
+     * CU Customization: Added method to check whether or not remittance information should be added to the payment.
+     */
+    private boolean shouldAddRemittanceInformationToPayment(
+            final PaymentGroup templatePaymentGroup,
+            final ExtractTypeContext extractTypeContext
+    ) {
+        if (extractTypeContext.isExtractionType(ExtractionType.CHECK)) {
+            return true;
+        } else if (extractTypeContext.isExtractionType(ExtractionType.ACH)) {
+            PayeeACHAccount payeeAchAccount = retrievePayeeACHAccount(templatePaymentGroup);
+            if (ObjectUtils.isNull(payeeAchAccount)) {
+                throw new IllegalStateException("Could not find Payee ACH Account for Payment Group '"
+                        + templatePaymentGroup.getId()  + "'; this should NEVER happen!");
+            }
+            return StringUtils.equals(payeeAchAccount.getStandardEntryClass(), StandardEntryClass.CTX.name());
+        } else {
+            throw new IllegalStateException("Extraction was for neither Check nor ACH; this should NEVER happen!");
```

Comment:
I would suggest adding the extraction type value to this error to help debug the problem if it ever happens

Reply:
Ok, I'll adjust the error message.